WordPress/gutenberg
# DataViews: add actions to list layout

Part of https://github.com/WordPress/gutenberg/issues/59659 and https://github.com/WordPress/gutenberg/issues/55083
Follow-up to https://github.com/WordPress/gutenberg/pull/59637
Fixes https://github.com/WordPress/gutenberg/issues/61555

## What?

This PR adds support for actions in list layout, which is only enabled for Pages and Templates as of now:

| Pages | Templates |
| --- | --- |
| <img width="1512" alt="Captura de ecrã 2024-05-10, às 11 49 29" src="https://github.com/WordPress/gutenberg/assets/583546/53bca771-e1b7-422e-af97-e8864027c615"> |  <img width="1512" alt="Captura de ecrã 2024-05-10, às 11 49 23" src="https://github.com/WordPress/gutenberg/assets/583546/f0f264c3-61cc-407b-b788-2781aec6cc24"> |

This PR also enables two-dimensional arrow-key navigation:

https://github.com/WordPress/gutenberg/assets/583546/81c24287-3f53-4744-bbc6-efd7948e0e2f

## Why?

We want to make them available. See https://github.com/WordPress/gutenberg/issues/59659

## How?

- Pass the existing `actions` prop that any other layout is using to the list view layout.
- Add two-dimensional arrow navigation to the list items, so users can use right/left arrow keys to move between the element and its actions.

## Testing Instructions

- Go to a page with list view (Pages or Templates) and verify the actions are present.
- Navigate via keyboard to verify that it works as expected:
  - The item list is a single tab stop.
  - You can vertically through the list via up/down arrow keys.
  - You can move horizontally within a single item via right/left arrow keys.
